src/completion/completeSnippets.ts

```diff
@@ -52,16 +52,36 @@ function doesStateSupportErrorHandling(node: ASTNode): boolean {
 
     return ERROR_HANDLING_STATES.includes(typeNode?.valueNode?.value as string)
 }
+interface CompleteSnippetsOptions {
+    shouldShowStateSnippets?: boolean,
+    shouldShowRetrySnippet?: boolean,
+    shouldShowCatchSnippet?: boolean
+}
 
-export default function completeSnippets(node: ASTNode | undefined, offset: number, shouldShowStateSnippets?: boolean): CompletionItem[] {
+export default function completeSnippets(node: ASTNode | undefined, offset: number, options?: CompleteSnippetsOptions): CompletionItem[] {
     if (node) {
+        const shouldShowErrorSnippets = options?.shouldShowCatchSnippet || options?.shouldShowRetrySnippet
         // If the value of shouldShowStateSnippets is false prevent the snippets from being displayed
-        if (shouldShowStateSnippets === undefined ? isChildOfStates(node) : shouldShowStateSnippets) {
+        if (options?.shouldShowStateSnippets === undefined ? isChildOfStates(node) : options.shouldShowStateSnippets) {
```

Comment:
nit: I am a bit concerned that using ternary operator within if/else statement hurts the general readability of code with limited gains, open for discussion, are there any eslint rule about this?

Reply:
+1, we should extract into a simple boolean to use in the condition